tests/integration/feature_store_integration/test_feature_materialize.py

```diff
@@ -0,0 +1,143 @@
+
+
+@pytest.fixture(name="deployed_feature_list", scope="module")
+def deployed_features_list_fixture(features):
+    """
+    Fixture for deployed feature list
+    """
+    feature_list = fb.FeatureList(features, name="EXTERNAL_FS_FEATURE_LIST")
+    feature_list.save()
+    with patch(
+        "featurebyte.service.feature_manager.get_next_job_datetime",
+        return_value=pd.Timestamp("2001-01-02 12:00:00").to_pydatetime(),
+    ):
+        deployment = feature_list.deploy()
+        deployment.enable()
+    yield deployment
+    deployment.disable()
+
+
+@pytest.fixture(name="default_feature_job_setting")
+def default_feature_job_setting_fixture(event_table):
+    """
+    Fixture for default feature job setting
+    """
+    return event_table.default_feature_job_setting
+
+
+@pytest.mark.parametrize("source_type", ["snowflake"], indirect=True)
+@pytest.mark.asyncio
+async def test_feature_materialize_service(
+    app_container,
+    session,
+    user_entity,
+    product_action_entity,
+    default_feature_job_setting,
+    deployed_feature_list,
+):
+    """
+    Test FeatureMaterializeService
+    """
+    _ = deployed_feature_list
+
+    service = app_container.feature_materialize_service
+    materialized_features = await service.materialize_features(
+        session=session,
+        primary_entity_ids=[user_entity.id],
+        feature_job_setting=default_feature_job_setting,
+    )
+
+    # Check offline store table for user entity
+    df = await session.execute_query(
+        f"SELECT * FROM {materialized_features.materialized_table_name}"
+    )
+    expected = [
+        "üser id",
+        "EXTERNAL_FS_AMOUNT_SUM_BY_USER_ID_24h_TIMES_100",
+        "EXTERNAL_FS_AMOUNT_SUM_BY_USER_ID_24h",
+        "__EXTERNAL_FS_COMPLEX_USER_X_PRODUCTION_ACTION_FEATURE__part0",
+    ]
+    assert set(df.columns.tolist()) == set(expected)
+    assert set(materialized_features.names) == set(expected[1:])
+    assert df.shape[0] == 9
+
+    # Check offline store table for product_action entity
+    materialized_features = await service.materialize_features(
+        session=session,
+        primary_entity_ids=[product_action_entity.id],
+        feature_job_setting=default_feature_job_setting,
+    )
+    df = await session.execute_query(
+        f"SELECT * FROM {materialized_features.materialized_table_name}"
+    )
+    expected = [
+        "PRODUCT_ACTION",
+        "__EXTERNAL_FS_COMPLEX_USER_X_PRODUCTION_ACTION_FEATURE__part1",
+    ]
+    assert set(df.columns.tolist()) == set(expected)
+    assert set(materialized_features.names) == set(expected[1:])
+    assert df.shape[0] == 5
+
+    # Check offline store table for combined entity (nothing to materialise here)
+    materialized_features = await service.materialize_features(
+        session=session,
+        primary_entity_ids=[user_entity.id, product_action_entity.id],
+        feature_job_setting=default_feature_job_setting,
+    )
+    assert materialized_features is None
```

Comment:
If i understood it correctly, if we add a feature that is grouped by `["ÜSER ID", "PRODUCT_ACTION"]`, we should expect some results here, right?

Reply:
Yes, that is right.